packages/core/src/modules/oob/repository/OutOfBandRecord.ts

```diff
@@ -0,0 +1,81 @@
+import type { TagsBase } from '../../../storage/BaseRecord'
+import type { DidCommService } from '../../dids'
+import type { OutOfBandRole } from '../domain/OutOfBandRole'
+import type { OutOfBandState } from '../domain/OutOfBandState'
+
+import { Type } from 'class-transformer'
+
+import { AriesFrameworkError } from '../../../error'
+import { BaseRecord } from '../../../storage/BaseRecord'
+import { uuid } from '../../../utils/uuid'
+import { OutOfBandMessage } from '../messages'
+
+export interface OutOfBandRecordProps {
+  id?: string
+  createdAt?: Date
+  updatedAt?: Date
+  tags?: TagsBase
+  outOfBandMessage: OutOfBandMessage
+  role: OutOfBandRole
+  state: OutOfBandState
+  autoAcceptConnection: boolean
+  reusable?: boolean
+}
+
+export class OutOfBandRecord extends BaseRecord<TagsBase> {
+  @Type(() => OutOfBandMessage)
+  public outOfBandMessage!: OutOfBandMessage
+  public role!: OutOfBandRole
+  public state!: OutOfBandState
+  public reusable!: boolean
+  public autoAcceptConnection!: boolean
+
+  public static readonly type = 'OutOfBandRecord'
+  public readonly type = OutOfBandRecord.type
+
+  public constructor(props: OutOfBandRecordProps) {
+    super()
+
+    if (props) {
+      this.id = props.id ?? uuid()
+      this.createdAt = props.createdAt ?? new Date()
+      this.outOfBandMessage = props.outOfBandMessage
+      this.role = props.role
+      this.state = props.state
+      this.autoAcceptConnection = props.autoAcceptConnection
+      this.reusable = props.reusable ?? false
+      this._tags = props.tags ?? {}
+    }
+  }
+
+  public getTags() {
+    const [recipientKey] = this.getRecipientKeys()
```

Comment:
We're only storing the first key here. Is this the key always used to send a message to?

Also, `recipientKey` feels a bit ambiguous to me. With the connection record we used `invitationKey` which was indepent of the role (just the key used for the invitation). But with `recipientKey` I'm wondering who's the recipient. As both sides are a recipient?

Reply:
Good point. I just did it in the same way we used `invitationKey` in connection record. We could change it to an array. Is it possible to search for records by tags containing array?

> Also, recipientKey feels a bit ambiguous to me.

I hear you, but I'm not sure if it helps. If it's `invitationKey` I still can wonder who's the invited, or not? It would definitely help to communicate that this key is part of oob invitation. On the other hand, we're introducing new terminology outside of protocol spec.